wcomponents-core/src/main/java/com/github/bordertech/wcomponents/render/webxml/WTextAreaRenderer.java

```diff
@@ -60,7 +58,10 @@ public void doRender(final WComponent component, final WebXmlRenderContext rende
 			xml.append("<ui:rtf />");
 		}
 
-		xml.appendEscaped(textArea.getText());
+		String textString = textArea.getText();

```

Comment:
To further clarify: anything which is not WTextArea with rich-text **and** read-only or is not WText **and** unencoded gets XML encoded. 

In all cases where XML encoding (actually escaping: another problem with our API) happens `&bul;` would become `&amp;bul;`. `&amp;` is XML safe, `&bul;` is not. Therefore we only need to worry about HTML character entities in the two cases mentioned above. We unescape HTML entities which are not XML entities because if we did not then these entities would cause XML failure (or we could be really perverse and add a doctype for **all** entitities 😉). 

I would prefer to get rid of XML completely but that is not an option in the time frames in which we **have** to deliver a means to make WTextArea in rich-text mode safe to implement.